lambdas/utils/lloyd_george_validator.py

```diff
@@ -140,31 +140,56 @@ def validate_filename_with_patient_details(
 ):
     try:
         file_name_info = extract_info_from_filename(file_name_list[0])
-        patient_name = file_name_info["patient_name"]
-        date_of_birth = file_name_info["date_of_birth"]
-
-        date_of_birth = datetime.datetime.strptime(date_of_birth, "%d-%m-%Y").date()
-        if patient_details.birth_date != date_of_birth:
-            raise LGInvalidFilesException("Patient DoB does not match our records")
-        patient_full_name = (
-            " ".join([name for name in patient_details.given_Name])
-            + " "
-            + patient_details.family_name
-        )
-        logger.info("Verifying patient name against the record in PDS...")
-
-        if not names_are_matching(patient_name, patient_full_name):
-            raise LGInvalidFilesException("Patient name does not match our records")
+        file_patient_name = file_name_info["patient_name"]
+        file_date_of_birth = file_name_info["date_of_birth"]
+        validate_patient_date_of_birth(file_date_of_birth, patient_details)
+        validate_patient_name(file_patient_name, patient_details)
 
     except (ClientError, ValueError) as e:
         logger.error(e)
         raise LGInvalidFilesException(e)
 
 
+def validate_patient_name(file_patient_name, pds_patient_details):
+    logger.info("Verifying patient name against the record in PDS...")
+    patient_name_split = file_patient_name.split(" ")
+    file_patient_first_name = patient_name_split[0]
+    file_patient_last_name = patient_name_split[-1]
+    is_file_first_name_in_patient_details = False
+    for patient_name in pds_patient_details.given_name:
+        if (
+            names_are_matching(file_patient_first_name, patient_name)
+            or not patient_name
+        ):
+            is_file_first_name_in_patient_details = True
+            break
```

Comment:
seems that this part will set the value to True when pds_patient_details.given_name array contains a falsey value (e.g. 0, empty string, None), is that on purpose?